From 433af014a60b48a87e5e5aa601acc54f960f6304 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Mon, 15 Feb 2016 15:31:52 +0800
Subject: [PATCH 3/3] atomisp2: css fix zero division issue

Change-Id: I2f0c810ec52312d4fc390d4e1df31b146f01fcac
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20625
Track-On: https://jira01.devtools.intel.com/browse/OAM-18671
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/101599
Reviewed-by: Bingbu Cao <bingbu.cao@intel.com>
Approver: Bingbu Cao <bingbu.cao@intel.com>
Reviewed-by: Vns Murthy Sristi <vns.murthy.sristi@intel.com>
Reviewed-by: Zaikuo Wang <zaikuo.wang@intel.com>
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/471918
---
 .../camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c     | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c
index d055c66..dff03f3 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c
@@ -114,7 +114,10 @@ crop_and_interpolate(unsigned int cropped_width,
 	out_cell_size = CEIL_DIV(padded_width, out_table->width - 1);
 	in_cell_size  = CEIL_DIV(sensor_width, table_width - 1);
 
-	assert(in_cell_size != 0);
+	if (0 == in_cell_size) {
+		assert(in_cell_size != 0);
+		return;
+	}
 
 	out_start_col = ((int)sensor_width - (int)cropped_width)/2 - left_padding;
 	out_start_row = ((int)sensor_height - (int)cropped_height)/2 - top_padding;
-- 
1.9.1

